subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java

```diff
@@ -52,7 +56,22 @@ protected CodeQualityExtension createExtension() {
         extension = project.getExtensions().create("checkstyle", CheckstyleExtension.class, project);
         extension.setToolVersion(DEFAULT_CHECKSTYLE_VERSION);
 
-        extension.setConfigDir(project.file("config/checkstyle"));
+        extension.setConfigurationDirectory(project.provider(new Callable<Directory>() {
+            @Override
+            public Directory call() {
+                DirectoryProperty configDir = project.getLayout().directoryProperty();
```

Comment:
:x: You don't want to create a property here, but a simple directory.

Reply:
done